src/apikey/index.ts

```diff
@@ -22,6 +23,8 @@ export async function withAPIKey(apiKey: string): Promise<SDKAuthHelper> {
           return requestToSign;
         },
       },
+      // Empty value to avoid calling the default credential providers chain
+      credentials: (async () => ({})) as AwsCredentialIdentityProvider,
```

Comment:
Thank you for the change! This makes a lot of sense.

That being said, I do have some curiosity here. 

Does this mean that this client configuration will only work with API keys now? As in the location client created in this fashion will not be usable with APIs that does not support API Keys. Is that a fair assumption?

Reply:
`signer` is the implementation detail that actually signs the request.

Since this method already overrode `signer` with a custom implementation, this means that any clients using `withAPIKey()` were only signing with API keys.

Reply:
Since this change apply to only the apikey part of the auth helper, the cognito one should not be affected.